dts/arm/st/n6/stm32n6.dtsi

```diff
@@ -266,7 +266,7 @@
 		};
 
 		pinctrl: pin-controller@56020000 {
-			compatible = "st,stm32-pinctrl";
+			compatible = "st,stm32n6-pinctrl";
```

Comment:
Optional: You can also chain the compatibles:
```
			compatible = "st,stm32-pinctrl", "st,stm32n6-pinctrl";
```
This simplifies a tiny bit the next part (Kconfig)